modelarmor/api/ModelArmor.Samples/CreateTemplateWithAdvancedSdp.cs

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2025 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+// [START modelarmor_create_template_with_advanced_sdp]
+using System;
+using Google.Api.Gax.ResourceNames;
+using Google.Cloud.Dlp.V2;
+using Google.Cloud.ModelArmor.V1;
+
+public class CreateTemplateWithAdvancedSdpSample
+{
+    public Template CreateTemplateWithAdvancedSdp(
+        string projectId = "my-project",
+        string locationId = "us-central1",
+        string templateId = "my-template",
+        string inspectTemplateId = "inspect-template-id",
+        string deidentifyTemplateId = "deidentify-template-id"
+    )
+    {
+        // Construct the API endpoint URL.
+        ModelArmorClientBuilder clientBuilder = new ModelArmorClientBuilder
+        {
+            Endpoint = $"modelarmor.{locationId}.rep.googleapis.com",
+        };
+
+        // Create the client.
+        ModelArmorClient client = clientBuilder.Build();
+
+        // Build the parent resource name.
+        LocationName parent = LocationName.FromProjectLocation(projectId, locationId);
+
+        // Build the inspect template name.
+        string inspectTemplateName = DeidentifyTemplateName
+            .FormatProjectLocationDeidentifyTemplate(projectId, locationId, inspectTemplateId)
+            .ToString();
+
+        // Build the deidentify template name.
+        string deidentifyTemplateName = DeidentifyTemplateName
+            .FormatProjectLocationDeidentifyTemplate(projectId, locationId, deidentifyTemplateId)
+            .ToString();
+
+        // Build the Model Armor template with Advanced SDP Filter.
+
+        // Note: If you specify only Inspect template, Model Armor reports the filter matches if
+        // sensitive data is detected. If you specify Inspect template and De-identify template, Model
+        // Armor returns the de-identified sensitive data and sanitized version of prompts or
+        // responses in the deidentifyResult.data.text field of the finding.
+        SdpAdvancedConfig advancedSdpConfig = new SdpAdvancedConfig
+        {
+            InspectTemplate = inspectTemplateName,
+            DeidentifyTemplate = deidentifyTemplateName,
```

Comment:
There should be an `InspectTemplateNameAsResourceName` and equivalent for the deidentify template. You should be able to use resource name helpers without converting them to strings.